### ISSUE TITLE
BUG: OrderedLogistic triggers CategoricalGibbsMetropolis sampler when unobserved

### ISSUE THREAD

**Issue by kyleabeauchamp** [pymc-devs/pymc#6816]
### Describe the issue:

If you take the example from https://www.pymc.io/projects/docs/en/stable/api/distributions/generated/pymc.OrderedLogistic.html and remove the observed data, the sampler switches from NUTS to CategoricalGibbs (and is much slower).

### Reproduceable code example:

```python
import numpy as np
import pymc as pm


# Generate data for a simple 1 dimensional example problem
n1_c = 300; n2_c = 300; n3_c = 300
cluster1 = np.random.randn(n1_c) + -1
cluster2 = np.random.randn(n2_c) + 0
cluster3 = np.random.randn(n3_c) + 2

x = np.concatenate((cluster1, cluster2, cluster3))
y = np.concatenate((1*np.ones(n1_c),
                    2*np.ones(n2_c),
                    3*np.ones(n3_c))) - 1

# Ordered logistic regression
with pm.Model() as model:
    cutpoints = pm.Normal("cutpoints", mu=[-1,1], sigma=10, shape=2,
                          transform=pm.distributions.transforms.univariate_ordered)
    y_ = pm.OrderedLogistic("y", cutpoints=cutpoints, eta=x)
    idata = pm.sample()
```

### Error message:
```
Multiprocess sampling (4 chains in 4 jobs)
CompoundStep
>NUTS: [cutpoints]
>CategoricalGibbsMetropolis: [y]
```
### PyMC version information:

In [4]: pm.__version__
Out[4]: '5.5.0'

**Comment by kyleabeauchamp** [comment]
I believe the OrderedLogistic probabilities should be continuous and nearly as smooth as other things that work with NUTS like Laplace distributions etc.

**Comment by kyleabeauchamp** [comment]
I can confirm that adding in the probabilities as a deterministic function of the logits (pasted below) does NOT trigger a reversion to Metropolis:

```
        c1 = cutpoints[:, 0]
        c2 = cutpoints[:, 1]

        p0 = 1 - pm.math.sigmoid(logits - c1)
        p1 = pm.math.sigmoid(logits - c1) - pm.math.sigmoid(logits - c2)
        p2 = pm.math.sigmoid(logits - c2)

        p1 = pm.Deterministic("p1", p1)
        probs = pm.Deterministic("probs", tt.as_tensor_variable([p0, p1, p2]))
```

**Comment by kyleabeauchamp** [comment]
I guess this might just make sense because it's a discrete distribution, but the probability component of this distribution is continuous and useful as a regression component...

**Comment by ricardoV94** [comment]
I guess you don't want to sample the categorical. If you just want the probabilities you can have them as the Deterministic you just created.

**Comment by kyleabeauchamp** [comment]
Yeah, the issue is that I want to have the probabilities from OrderedLogistic for use in a weighted logp via the `pm.logp(var, observed)` construct, but this is impossible as-is because instantiating the OrderedLogistic variable slows down the sampler it seems...

**Comment by ricardoV94** [comment]
I don't see why you need the `OrderedLogistic`?

**Comment by kyleabeauchamp** [comment]
So I am using the OrderedLogistic functional form of the probabilities---eta and cutpoints.  But it seems like I'll just need to rewrite it as a deterministic.

**Comment by ricardoV94** [comment]
If you want the probabilities you can do 
```
probs = pm.OrderedLogistic.dist(...).owner.inputs[-1]
```
And then wrap them in a `Deterministic`
